### PR TITLE
[LibOS] test/regression: Use unresponsive IPv4 address `203.0.113.0`

### LINKED ISSUE #1850: LibOS test: tcp_einprogress related testcases fails intermittently across distros
### Description of the problem

test_308_socket_tcp_einprogress_unresponsive_epoll and test_307_socket_tcp_einprogress_unresponsive_poll fails intermittently with Gramine Native & SGX

The issue is intermittent, we had run 2 nightly's but the issue is only seen in yesterday's nightly.
The failure is seen across all the distro's, and reproducible with both Gramine-Direct and SGX, but occurrence with Gramine-Direct is higher. In SGX, it can be seen sometimes 1 out of 6 times, but in Gramine-Direct it is 1 out of 2 times

### Steps to reproduce

1. meson setup build/ --buildtype=release -Ddirect=enabled -Dsgx=enabled -Dtests=enabled
2. ninja -C build
3. sudo ninja -C build install
4. cd libos/test/regression/
5. Run the below bash script ./test_epoll.sh

bash script: test_epoll.sh
#!/bin/bash
set -e

for i in {0..50}
do
       echo "Iteration: ", ${i}
       SGX=1 python3 -m pytest -v -s -k test_308_socket_tcp_einprogress_unresponsive_epoll
done

### Expected results

**Gramine SGX:**
```
test_libos.py::TC_80_Socket::test_308_socket_tcp_einprogress_unresponsive_epoll [0.001] Gramine is starting. Parsing TOML manifest file, this may take some time...
[0.675] -----------------------------------------------------------------------------------------------------------------------
[0.675] Gramine detected the following insecure configurations:
[0.675]
[0.675]   - sgx.debug = true                           (this is a debug enclave)
[0.675]   - loader.insecure__use_cmdline_argv = true   (forwarding command-line args from untrusted host to the app)
[0.675]   - sys.experimental__enable_flock = true      (flock syscall is enabled; still under development and may contain bugs)
[0.675]   - sgx.allowed_files = [ ... ]                (some files are passed through from untrusted host without verification)
[0.675]   - fs.insecure__keys.* = "..."                (keys hardcoded in manifest)
[0.675]
[0.675] Gramine will continue application execution, but this configuration must not be used in production!
[0.675] -----------------------------------------------------------------------------------------------------------------------
[0.675]
[0.723] local address 10.66.254.169:34120
[1.724] TEST OK (connection timed out)
```

**Gramine-Direct:**
```
test_libos.py::TC_80_Socket::test_308_socket_tcp_einprogress_unresponsive_epoll [0.021] local address 10.66.254.169:37002
[1.022] TEST OK (connection timed out)
```

### Actual results

**Gramine SGX:**
```
test_libos.py::TC_80_Socket::test_308_socket_tcp_einprogress_unresponsive_epoll [0.001] Gramine is starting. Parsing TOML manifest file, this may take some time...
[0.673] -----------------------------------------------------------------------------------------------------------------------
[0.673] Gramine detected the following insecure configurations:
[0.673]
[0.673]   - sgx.debug = true                           (this is a debug enclave)
[0.674]   - loader.insecure__use_cmdline_argv = true   (forwarding command-line args from untrusted host to the app)
[0.674]   - sys.experimental__enable_flock = true      (flock syscall is enabled; still under development and may contain bugs)
[0.674]   - sgx.allowed_files = [ ... ]                (some files are passed through from untrusted host without verification)
[0.674]   - fs.insecure__keys.* = "..."                (keys hardcoded in manifest)
[0.674]
[0.674] Gramine will continue application execution, but this configuration must not be used in production!
[0.674] -----------------------------------------------------------------------------------------------------------------------
[0.674]
[0.731] local address 10.66.254.169:33830
[1.495] TEST OK (connection refused after initial EINPROGRESS)
```

**Gramine Direct:**
```
test_libos.py::TC_80_Socket::test_308_socket_tcp_einprogress_unresponsive_epoll [0.021] local address 10.66.254.169:48686
[0.876] TEST OK (connection refused after initial EINPROGRESS)
FAILED
```

Logs are attached:
[socket_tcp_einprogress_unresponsive_epoll.log](https://github.com/gramineproject/gramine/files/15020133/socket_tcp_einprogress_unresponsive_epoll.log)
[socket_tcp_einprogress_unresponsive_epoll_gdirect.log](https://github.com/gramineproject/gramine/files/15020134/socket_tcp_einprogress_unresponsive_epoll_gdirect.log)

### Gramine commit hash

c59c041cc4b1c010b9036af87c58c8cacad5eb9c

### PR DESCRIPTION
**Pull request by dimakuv**
## Description of the changes 

Previously the EINPROGRESS tests used a supposedly-unresponsive address `10.255.255.255`, but this address may be used in the private network. This commit switches to the address from the reserved "Documentation" range, which should never be used in networks.

Fixes #1850.

## How to test this PR? 

Run in a network that either actually uses 10.255.255.255, or that explicitly refuses connections to it.

### CODE REVIEW

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: 0 of 2 files reviewed, all discussions resolved, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1851#-NvkBnlN39WrxYWay1rG:-NvkBnlN39WrxYWay1rH:b-79o5o3) (no related file):*
FYI: The IP address is taken from the range found in https://en.wikipedia.org/wiki/Reserved_IP_addresses and recommended by @woju 

___

**Review by woju — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 3 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @dimakuv)

___
*[`libos/test/regression/tcp_einprogress.c` line 26 at r1](https://reviewable.io/reviews/gramineproject/gramine/1851#-NvkC2ps7f6PKQWe6HUP:-NvkC2ps7f6PKQWe6HUQ:b-4i91id) ([raw file](https://github.com/gramineproject/gramine/blob/07fb72055340bcebf9e275af771160ee79410c24/libos/test/regression/tcp_einprogress.c#L26)):*
> ```C
> static void usage(const char* prog_name) {
>     fprintf(stderr, "usage: %s <IP address> poll|epoll\n", prog_name);
>     fprintf(stderr, "(use 127.0.0.1 for responsive peer and 203.0.113.0 for unresponsive peer)\n");
> ```

This should not be `.0` address, because that technically is a network address, not a host address. Can you change to `.1` or any other number that is not `.0` nor `.255`? 
___
*[`libos/test/regression/test_libos.py` line 1485 at r1](https://reviewable.io/reviews/gramineproject/gramine/1851#-NvkBp3D8Gw81xsScj8y:-NvkBp3D8Gw81xsScj8z:b-7ropln) ([raw file](https://github.com/gramineproject/gramine/blob/07fb72055340bcebf9e275af771160ee79410c24/libos/test/regression/test_libos.py#L1485)):*
> ```Python
>     # times out because the connection cannot be established
>     def test_307_socket_tcp_einprogress_unresponsive_poll(self):
>         stdout, _ = self.run_binary(['tcp_einprogress', '203.0.113.0', 'poll'])
> ```

ditto
___
*[`libos/test/regression/test_libos.py` line 1489 at r1](https://reviewable.io/reviews/gramineproject/gramine/1851#-NvkC-1R5ewCA9fb01UW:-NvkC-1R5ewCA9fb01UX:b-7ropln) ([raw file](https://github.com/gramineproject/gramine/blob/07fb72055340bcebf9e275af771160ee79410c24/libos/test/regression/test_libos.py#L1489)):*
> ```Python
> 
>     def test_308_socket_tcp_einprogress_unresponsive_epoll(self):
>         stdout, _ = self.run_binary(['tcp_einprogress', '203.0.113.0', 'epoll'])
> ```

ditto

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: 0 of 2 files reviewed, 3 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @woju)

___
*[`libos/test/regression/tcp_einprogress.c` line 26 at r1](https://reviewable.io/reviews/gramineproject/gramine/1851#-NvkC2ps7f6PKQWe6HUP:-NvkDBe6Ey1wRi8jbEIi:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/07fb72055340bcebf9e275af771160ee79410c24/libos/test/regression/tcp_einprogress.c#L26)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

This should not be `.0` address, because that technically is a network address, not a host address. Can you change to `.1` or any other number that is not `.0` nor `.255`?
</blockquote></details>

Done.
___
*[`libos/test/regression/test_libos.py` line 1485 at r1](https://reviewable.io/reviews/gramineproject/gramine/1851#-NvkBp3D8Gw81xsScj8y:-NvkDC64BUobgN2XFq4L:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/07fb72055340bcebf9e275af771160ee79410c24/libos/test/regression/test_libos.py#L1485)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.
___
*[`libos/test/regression/test_libos.py` line 1489 at r1](https://reviewable.io/reviews/gramineproject/gramine/1851#-NvkC-1R5ewCA9fb01UW:-NvkDCTG6aiuyTMil8Io:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/07fb72055340bcebf9e275af771160ee79410c24/libos/test/regression/test_libos.py#L1489)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

**Review by woju — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required)

**Review by kailun-qin — DISMISSED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 5 unresolved discussions (waiting on @dimakuv)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwX6fN_EvypKpBWTdF8:-NwX6fN_EvypKpBWTdF9:b-psucuu) (no related file):*
> Run in a network that either actually uses 10.255.255.255, or that explicitly refuses connections to it.

Huh, what network does this? The latter is probably some very aggressive firewall, but the former?

___
___
*[`-- commits` line 5 at r2](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwX6cCk4IZjErxwWXn3):*




_Suggestion:_
```
in a private network.
```
___
*[`libos/test/regression/tcp_einprogress.c` line 26 at r2](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwX7KU-CxVhw1sFwmOJ:-NwX7KU-CxVhw1sFwmOK:bwefvtj) ([raw file](https://github.com/gramineproject/gramine/blob/65b35905544bc18ae607bc61645c5b50c844ac31/libos/test/regression/tcp_einprogress.c#L26)):*
> ```C
> static void usage(const char* prog_name) {
>     fprintf(stderr, "usage: %s <IP address> poll|epoll\n", prog_name);
>     fprintf(stderr, "(use 127.0.0.1 for responsive peer and 203.0.113.1 for unresponsive peer)\n");
> ```

Please add a short comment that this address is from a reserved "Documentation" range. Otherwise it looks quite random, like hardcoding an address of some real service on the internet.
___
*[`libos/test/regression/test_libos.py` line 1485 at r2](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwX7BpE0tOMt6aN7x-V:-NwX7BpE0tOMt6aN7x-W:b-7ropln) ([raw file](https://github.com/gramineproject/gramine/blob/65b35905544bc18ae607bc61645c5b50c844ac31/libos/test/regression/test_libos.py#L1485)):*
> ```Python
>     # times out because the connection cannot be established
>     def test_307_socket_tcp_einprogress_unresponsive_poll(self):
>         stdout, _ = self.run_binary(['tcp_einprogress', '203.0.113.1', 'poll'])
> ```

ditto
___
*[`libos/test/regression/test_libos.py` line 1489 at r2](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwX7GRR2Wa9_VjomHb1:-NwX7GRR2Wa9_VjomHb2:b-7ropln) ([raw file](https://github.com/gramineproject/gramine/blob/65b35905544bc18ae607bc61645c5b50c844ac31/libos/test/regression/test_libos.py#L1489)):*
> ```Python
> 
>     def test_308_socket_tcp_einprogress_unresponsive_epoll(self):
>         stdout, _ = self.run_binary(['tcp_einprogress', '203.0.113.1', 'epoll'])
> ```

ditto

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 6 unresolved discussions (waiting on @mkow)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwX6fN_EvypKpBWTdF8:-NwcvwsxA2OJ6HPVHo77:b-x44knu) (no related file):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

> Run in a network that either actually uses 10.255.255.255, or that explicitly refuses connections to it.

Huh, what network does this? The latter is probably some very aggressive firewall, but the former?
</blockquote></details>

I don't know about the former, maybe nobody ever uses it (but like why not?)

We hit this problem in a private corporate network, where indeed there is an aggressive firewall. So we hit the latter case.

___
*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwcwC-z0DaDaFkHNHCV:-NwcwC-z0DaDaFkHNHCW:b-2lw3fv) (no related file):*
I'm thinking of closing this PR, because the new address `203.0.113.1` is equally susceptible to the problem of an aggressive firewall (that explicitly refuses connections to this address).

So I don't know, feel like there's no good way to satisfy all firewalls in CIs/environments where Gramine is tested...

___

**Review by mkow — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 6 unresolved discussions (waiting on @dimakuv)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwcwC-z0DaDaFkHNHCV:-NwdU0pM19OG6etErLWZ:b-xj5ec2) (no related file):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

I'm thinking of closing this PR, because the new address `203.0.113.1` is equally susceptible to the problem of an aggressive firewall (that explicitly refuses connections to this address).

So I don't know, feel like there's no good way to satisfy all firewalls in CIs/environments where Gramine is tested...
</blockquote></details>

Yeah, I'm not a fan of this change, but I'm also ok with merging if it would actually solve a problem (but seems it doesn't).

___

**Review by woju — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 6 unresolved discussions (waiting on @dimakuv and @mkow)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwcwC-z0DaDaFkHNHCV:-NweOuSY5aST5nVRnQpl:b-xti84u) (no related file):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Yeah, I'm not a fan of this change, but I'm also ok with merging if it would actually solve a problem (but seems it doesn't).
</blockquote></details>

I've made our CI drop TEST-NET-1, 2 and 3 (also IPv6: 2001:db8::/32 doc prefix and 100::/64 discard prefix). This PR is really the correct solution, because 10/8 can be used by whatever external to us networks and we can do nothing about it. But TEST-NET is fair game, we can declare that whoever runs this needs silent address from any of the TEST-NETs (in contrast to 10/8: we can't arbitrarily declare any requirements against 10/8).

___

**Comment by dimakuv** [comment]
Jenkins, test this please

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 6 unresolved discussions (waiting on @mkow and @woju)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwcwC-z0DaDaFkHNHCV:-NwhYwiB72j5WfDJHvRf:b-itn5ff) (no related file):*
> I've made our CI drop TEST-NET-1, 2 and 3

What do you mean by "drop"? We actually don't want to drop (immediately reject connection attempts), we want these addresses to be "unresponsive" (be rejected after some time, after the connection timeout in Linux). Is that what you meant?

I restarted the CI to see if this PR passes now. Before it was immediately rejecting connection attempts, which means that our LibOS test fails.

___

**Review by woju — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 6 unresolved discussions (waiting on @dimakuv and @mkow)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwcwC-z0DaDaFkHNHCV:-NxCGcPoBp9nXA_T9IyL:bfkv0gf) (no related file):*
> What do you mean by "drop"? We actually don't want to drop (immediately reject connection attempts)

"DROP" and "REJECT" have specific meaning in the context of the linux network stack (incl. netfilter aka "nftables/iptables" aka "firewall"):

https://manpages.debian.org/bookworm/nftables/nft.8.en.html#VERDICT_STATEMENT
https://manpages.debian.org/bookworm/nftables/nft.8.en.html#REJECT_STATEMENT

This terminology BTW dates back to at least ipchains (late 1990's): https://tldp.org/HOWTO/IPCHAINS-HOWTO-4.html#target-spec, but is probably older, possibly taken up from ipfirewall, yet older linux packet filter ported from BSD world.

So I made our CI "drop" the packets, which results in silence as observed by network stack attached to the process that would like to `connect(2)` to this address (no one will receive SYN packets, so no one will answer them). The previous behaviour was most likely "reject" or something comparable.

___

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: 0 of 2 files reviewed, 5 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel), "fixup! " found in commit messages' one-liners (waiting on @dimakuv, @kailun-qin, @mkow, and @woju)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwcwC-z0DaDaFkHNHCV:-NxCYaKeDqH579ogDQ7q:bp5xn0f) (no related file):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

> What do you mean by "drop"? We actually don't want to drop (immediately reject connection attempts)

"DROP" and "REJECT" have specific meaning in the context of the linux network stack (incl. netfilter aka "nftables/iptables" aka "firewall"):

https://manpages.debian.org/bookworm/nftables/nft.8.en.html#VERDICT_STATEMENT
https://manpages.debian.org/bookworm/nftables/nft.8.en.html#REJECT_STATEMENT

This terminology BTW dates back to at least ipchains (late 1990's): https://tldp.org/HOWTO/IPCHAINS-HOWTO-4.html#target-spec, but is probably older, possibly taken up from ipfirewall, yet older linux packet filter ported from BSD world.

So I made our CI "drop" the packets, which results in silence as observed by network stack attached to the process that would like to `connect(2)` to this address (no one will receive SYN packets, so no one will answer them). The previous behaviour was most likely "reject" or something comparable.
</blockquote></details>

@woju Thanks for the explanations and for changing our CI! The PR passes CI now.

___
___
*[`-- commits` line 5 at r2](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwX6cCk4IZjErxwWXn3):*
Will do during final rebase
___
*[`libos/test/regression/tcp_einprogress.c` line 26 at r2](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwX7KU-CxVhw1sFwmOJ:-NxCapP4CjKVhZGtXQUd:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/65b35905544bc18ae607bc61645c5b50c844ac31/libos/test/regression/tcp_einprogress.c#L26)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Please add a short comment that this address is from a reserved "Documentation" range. Otherwise it looks quite random, like hardcoding an address of some real service on the internet.
</blockquote></details>

Done.
___
*[`libos/test/regression/test_libos.py` line 1485 at r2](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwX7BpE0tOMt6aN7x-V:-NxCb5sd48nKyLU4CJua:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/65b35905544bc18ae607bc61645c5b50c844ac31/libos/test/regression/test_libos.py#L1485)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.
___
*[`libos/test/regression/test_libos.py` line 1489 at r2](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwX7GRR2Wa9_VjomHb1:-NxCb6Qx4-hynLPQBcHM:b-896fix) ([raw file](https://github.com/gramineproject/gramine/blob/65b35905544bc18ae607bc61645c5b50c844ac31/libos/test/regression/test_libos.py#L1489)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

**Review by woju — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 5 unresolved discussions, "fixup! " found in commit messages' one-liners (waiting on @dimakuv and @mkow)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwcwC-z0DaDaFkHNHCV:-NxD4H9a9U3DUOMTs-Tx:bfkqs9j) (no related file):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

@woju Thanks for the explanations and for changing our CI! The PR passes CI now.
</blockquote></details>

Anytime. That's good to hear.

___

**Review by woju — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 6 unresolved discussions, "fixup! " found in commit messages' one-liners (waiting on @dimakuv and @mkow)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwcwC-z0DaDaFkHNHCV:-NxHGOD92ZixGq5umAo3:b-zdmhfp) (no related file):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

> I've made our CI drop TEST-NET-1, 2 and 3 (also IPv6: 2001:db8::/32 doc prefix and 100::/64 discard prefix).

Wait, does this mean that you need to specially modify your system to make the test working? I.e. when run normally (i.e. a random user with stock ubuntu and no special net config) the test will fail?
</blockquote></details>

On most systems and networks, no, you don't need to do anything. But if you have insane network, then your box sometimes need to correct for network admin. Another solution would be for us to buy a public IP address and publicly document it as blackhole for all people to sink their traffic there. I don't want to maintain that. I don't see any other viable solutions.

Longer explanation is, user should know him/herself, what addresses on his/her network aren't routed and pick from that. In our CI, I guarantee that TEST-NET-123 are dropped. On other networks people should be free to pick another address if someone else took possession of TEST-NETs, so the unconditional explanation "use X here" might or might not be misleading. I'd reword it to something like "use IP address that your network does not route, you can try 203.0.113.1 if you don't know what you're doing".

___

**Review by mkow — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 6 unresolved discussions, "fixup! " found in commit messages' one-liners (waiting on @dimakuv)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwcwC-z0DaDaFkHNHCV:-NxIGS9a4wG9gbYD6-ck:b-9uw76y) (no related file):*
> I've made our CI drop TEST-NET-1, 2 and 3 (also IPv6: 2001:db8::/32 doc prefix and 100::/64 discard prefix).

> On most systems and networks, no, you don't need to do anything.

Then why did you have to drop these addresses explicitly in our CI? That's the part I don't get and it sounded like you needed to do something special to make these new versions of the tests work.
So, you did that dropping only to ensure that it's actually dropped? And it wasn't strictly needed?

___

**Review by woju — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 6 unresolved discussions, "fixup! " found in commit messages' one-liners (waiting on @dimakuv and @mkow)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwcwC-z0DaDaFkHNHCV:-NxIaczH4OgLzZURd_xB:b1wbvs0) (no related file):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

> I've made our CI drop TEST-NET-1, 2 and 3 (also IPv6: 2001:db8::/32 doc prefix and 100::/64 discard prefix).

> On most systems and networks, no, you don't need to do anything.

Then why did you have to drop these addresses explicitly in our CI? That's the part I don't get and it sounded like you needed to do something special to make these new versions of the tests work.
So, you did that dropping only to ensure that it's actually dropped? And it wasn't strictly needed?
</blockquote></details>

It was needed, because UNC network REJECTs TEST-NETs. It is the case that I need to correct for Uni's admin.

___

**Review by mkow — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 2 unresolved discussions, "fixup! " found in commit messages' one-liners (waiting on @dimakuv)

**Review by kailun-qin — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 2 unresolved discussions, "fixup! " found in commit messages' one-liners

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 2 unresolved discussions (waiting on @mkow)

___
*[`-- commits` line 5 at r2](https://reviewable.io/reviews/gramineproject/gramine/1851#-NwX6cCk4IZjErxwWXn3):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Will do during final rebase
</blockquote></details>

Done.

**Review by kailun-qin — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: all files reviewed, 2 unresolved discussions (waiting on @mkow)

**Review by mkow — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1851)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved
